src/server.rs

```diff
@@ -1266,7 +1265,7 @@ where
                 }
             };
             let send = Box::pin(
-                tx.send(Ok(Response::CompileFinished(res))).map_err(|e| anyhow!("send on finish failed") )
+                tx.send(Ok(Response::CompileFinished(res))).map_err(|_e| anyhow!("send on finish failed") )
```

Comment:
We probably should attempt to attach `e` with `.context(e)` to the newly created error (or vice versa)

Reply:
On the other hand, my initial plan was to use `thiserror` on a per module basis.

Reply:
Thank you! I started with  `.context(e)` for now, but can try later to refactor into something else.

Done in ea50e18